opm/simulators/flow/BlackoilModel.hpp

```diff
@@ -272,7 +272,8 @@ namespace Opm {
             Dune::Timer perfTimer;
             perfTimer.start();
             // update the solution variables in the model
-            if ( timer.lastStepFailed() ) {
+            int lastStepFailed = grid_.comm().min(timer.lastStepFailed());
```

Comment:
Merging this change is problematic. If the processors do not agree on this then something is usually wrong and there is a bug. This change will further obfuscate this.

Can you please check whether this change is still needed or the processes now agree?

On the other hand the situation that the program continues computation if something is wrong is also not good. So maybe we should throw if there is no agreement:
```suggestion
            const int lastStepFailed = timer.lastStepFailed();
            if ( lastStepFailed != grid_.comm().min(lastStepFailed) {
              // throw / log an error...
            }
```
Maybe stuff for another separate PR, though?

Reply:
I checked that once more, it is not needed anymore, yet I have included a check here as suggested by @blattms, so we throw an error here in case we ever have a simulation in the future where things are not going correctly.